src/components/UserPage/UserPage.js

```diff
@@ -1,10 +1,135 @@
-import React, { Component } from 'react';
+import React, { Component, useEffect } from 'react';
 import { connect } from 'react-redux';
 import LogOutButton from '../LogOutButton/LogOutButton';
 import mapStoreToProps from '../../redux/mapStoreToProps';
 
+const mapStateToProps = (state, ownProps) => ({
+  dispatch: ownProps.dispatch,
+  history: ownProps.history,
+  user: state.user,
+  friends: state.friendsReducer,
+  decks: state.deck,
+})
```

Comment:
I like declaring mapStateToProps at the top. Even though the connect is at the bottom. This is so I can clearly see where all arguments for the component are coming from. The article I shared will help explain why you need to structure your return. We do not want our component to care about the state of the entire store. Only the pieces used by our component.
Notice my changes in naming also. You have some naming convention issues. Naming is one of the hardest things in programing.